packages/core/src/provider/types.ts

```diff
@@ -104,7 +105,7 @@ export interface ISmartAccountProvider<
   readonly feeDataGetter: AccountMiddlewareFn;
   readonly customMiddleware?: AccountMiddlewareFn;
 
-  readonly account?: ISmartContractAccount;
+  readonly account?: ISmartContractAccount<SignerClient>;
```

Comment:
can we get rid of the generic if we just make the smart account interface default to `any`?

Reply:
wouldn't we want the Client generic here since it informs what type the account's owner is?